src/analysis/special_functions/gamma.lean

```diff
@@ -0,0 +1,671 @@
+import analysis.special_functions.exponential
+import measure_theory.integral.integral_eq_improper
+import measure_theory.integral.limit_comparison
+import analysis.special_functions.integrals
+
+/-!
+# The Gamma function
+
+In this file we define the Γ function (of a real variable in the range `1 ≤ s`), using the
+definition `Γ(s) = ∫ x in 0..∞, x^(s-1) exp(-x) dx`, and prove that it satisfies the relation
+`Γ(s+1) = s Γ(s)`. We also prove that `Γ(n+1) = n!` for `n ∈ ℕ`. We also show that `Γ(s)` is
+continuous on `(0, ∞)` (but not yet at the left endpoint).
+
+TO DO:
+
+- Extend to the whole real line.
+- Allow complex `s` and prove analyticity.
+
+## Tags
+
+gamma
+-/
+
+noncomputable theory
+open finset filter metric interval_integral set function
+open_locale classical topological_space ennreal filter measure_theory
+
+namespace real
+
+/- The integrand function and its properties -/
+
+def F (s x : ℝ) : ℝ := exp(-x) * x^s
+
+
+/- We prove some lemmas about F:
+
+- for any `s ≥ 0`, F is continuous of `x ∈ [0,∞)`;
+- for any `s ≥ 1`, the derivative of `F(s, -)` at any `x ∈ [0, ∞)` is what it should be;
+- for any `s ≥ 1`, the derivative of `F(s, -)` is integrable on [0, X] for any X ≥ 0;
+- for any `s ∈ ℝ`, F is `O( exp(-(1/2) * x))` as `x → ∞`.
+-/
+
+lemma cont_F (s : ℝ) (h1: 0 ≤ s) :
+  continuous_on s.F (Ici 0) :=
+begin
```

Comment:
You may be able to prove this with `continuity`

Reply:
Thanks for the very useful suggestion -- I have now used this to save work in a bunch of other places, but actually not in this particular one: the problem is that the goal here is a `continuous_on`, not a `continuous`. There doesn't seem to be a `continuity_on` tactic!

Reply:
I tried it and it didn't seem to be able to :-(